Formula/v8.rb

```diff
@@ -3,110 +3,90 @@
-  end
-
-  resource "common" do
-    url "https://chromium.googlesource.com/chromium/src/base/trace_event/common.git",
-        :revision => "c8c8665c2deaf1cc749d9f8e153256d4f67bf1b8"
-  end
-
-  resource "swarming_client" do
-    url "https://chromium.googlesource.com/external/swarming.client.git",
-        :revision => "df6e95e7669883c8fe9ef956c69a544154701a49"
-  end
-
-  resource "gtest" do
-    url "https://chromium.googlesource.com/external/github.com/google/googletest.git",
-        :revision => "6f8a66431cb592dad629028a50b3dd418a408c87"
-  end
-
-  resource "gmock" do
-    url "https://chromium.googlesource.com/external/googlemock.git",
-        :revision => "0421b6f358139f02e102c9c332ce19a33faf75be"
-  end
-
-  resource "clang" do
-    url "https://chromium.googlesource.com/chromium/src/tools/clang.git",
-        :revision => "faee82e064e04e5cbf60cc7327e7a81d2a4557ad"
+  resource "depot_tools" do
+    url "https://chromium.googlesource.com/chromium/tools/depot_tools.git",
+        :revision => "dbbf350a34ab7991e635f91a70bd2d211eb921be"
   end
 
   def install
-    # Bully GYP into correctly linking with c++11
-    ENV.cxx11
-    ENV["GYP_DEFINES"] = "clang=1 mac_deployment_target=#{MacOS.version}"
-    # https://code.google.com/p/v8/issues/detail?id=4511#c3
-    ENV.append "GYP_DEFINES", "v8_use_external_startup_data=0"
-
-    # fix up libv8.dylib install_name
-    # https://github.com/Homebrew/homebrew/issues/36571
-    # https://code.google.com/p/v8/issues/detail?id=3871
-    inreplace "tools/gyp/v8.gyp",
-              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
-              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
-
-    (buildpath/"build/gyp").install resource("gyp")
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    (buildpath/"depot_tools").install resource("depot_tools")
+    ENV.prepend_path "PATH", buildpath/"depot_tools"
+
+    # This env variable used by gclient to prevent depot_tools to update depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    repo_cache = HOMEBREW_CACHE/"#{name}--v8--gclient/"
```

Comment:
This feels like a misuse of `HOMEBREW_CACHE`, to be honest. I'll let ILZ chime in though. We might simply be better using `buildpath`.

Reply:
I'll wait for more comments on this. I haven't edited a formula before, so I don't really know what the best practices are. 😅

Reply:
We do need some kind of caching mechanism here, but probably best not to do the build in the cache. I guess we could copy everything out of the cache into the buildpath after running glient sync.

Reply:
I just removed the caching entirely for now, as it didn't actually work properly--reinstalling would fail because the cache was dirty. I'll have to figure out if there's a reasonable way to cache. Unfortunately doing a `gclient sync` to the version you are already synced to will fail, so it's not as straightforward as just copy/pasting the directory. 🤔